### DISCUSSION TITLE
[Question] Bundle as a single executable binary?

### DISCUSSION THREAD

**Discussion by aechegoyan17** [danny-avila/LibreChat#2601] [Q&A]
Docker images are getting quite large and I would like to reduce their size by using a distroless image containing the bundled binary, which could also improve security as minimal dependencies would be bundled.

I was trying with vercel pkg without success due to module-alias. Also, pkg is no longer maintained and has been archived due to Node's new feature (https://nodejs.org/api/single-executable-applications.html).

Has anyone tried it?

**Comment by danny-avila** [comment]
I've not tried this, maybe this can help? 

https://ishu-harsh.medium.com/creating-an-executable-of-a-node-js-express-application-with-webpack-and-docker-854f97a7cca9
